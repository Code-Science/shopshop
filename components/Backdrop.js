import PropTypes from 'prop-types';
import styles from '../styles/components/Backdrop.module.css';

const Backdrop = ({ show, click }) =>
  show ? <div className={styles.Backdrop} onClick={click}></div> : null;

Backdrop.propTypes = {
  show: PropTypes.bool.isRequired,
  click: PropTypes.func,
};

export default Backdrop;
