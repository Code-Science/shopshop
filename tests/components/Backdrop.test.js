import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Backdrop from '../../components/Backdrop';

jest.mock('../../store/StateProvider');

configure({ adapter: new Adapter() });

describe('Backdrop Component', () => {
  const wrapper = shallow(<Backdrop show={true} />);

  it('expect to render Backdrop component', () => {
    expect(wrapper.length).toEqual(1);
  });
  it('snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
