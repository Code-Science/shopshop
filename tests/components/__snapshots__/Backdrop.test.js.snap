// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop Component snapshot test 1`] = `
<div
  className="Backdrop"
/>
`;
